src/array.c

```diff
@@ -627,15 +628,15 @@ static int NOINLINE array_resize_buffer(jl_array_t *a, size_t newlen)
         a->data = jl_gc_managed_realloc(olddata, nbytes, oldnbytes,
                                         a->flags.isaligned, (jl_value_t*)a);
     }
-    else if (a->flags.how == 3 && jl_is_string(jl_array_data_owner(a))) {
+    else if (a->flags.how == 3 && jl_is_string(jl_array_data_owner(a)) && !is_discriminated_union) {
```

Comment:
Shouldn't `jl_is_string(jl_array_data_owner(a))` rule out the possibility of `is_discriminated_union`? Were you seeing cases where isbits Union arrays hit this code path?

Reply:
Ah, yes, I think you're right.  No, the issue manifested itself via segfaults within the GC, so we didn't actually see this code path get taken for an isbits union — we just wanted to ensure that it wouldn't get taken for them.